File: UniEatsBackEnd/Models/Order.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Threading.Tasks;

namespace UniEatsBackEnd.Models
{
    public class Order
    {
        public int OrderId { get; set; }

        [Required]
        public int UserId { get; set; }

        public DateTime OrderDate { get; set; } = DateTime.Now;

        [Range(0.01, double.MaxValue, ErrorMessage = "Total amount must be greater than or equal to 0.")]
        public decimal TotalAmount { get; set; }

        [RegularExpression(@"^(Pending|Completed|Canceled)$", ErrorMessage = "Status must be 'Pending', 'Completed', or 'Canceled'.")]
        public string Status { get; set; } = "Pending";

        [RegularExpression(@"^(Card|Cash|Online|GiftCard)$", ErrorMessage = "Payment method must be one of: 'Card', 'Cash', 'Online', 'GiftCard'.")]
        public string PaymentMethod { get; set; }

        [RegularExpression(@"^(Pickup|Delivery)$", ErrorMessage = "Delivery method must be either 'Pickup' or 'Delivery'.")]
        public string DeliveryMethod { get; set; }

        [StringLength(255)]
        public string DeliveryAddress { get; set; }

        [StringLength(255)]
        public string OrderNotes { get; set; }

        // Navigation property (User)
        public User? User { get; set; }
    }
}

File: UniEatsFrontEnd/UniEatsFrontEnd/src/models/order.ts
//import { User } from './user.model';

export interface Order {
  orderId: number; // Unique identifier for the order
  userId: number; // ID of the user who placed the order
  orderDate?: Date; // Defaults to the current date
  totalAmount: number; // Total amount for the order, must be >= 0.01
  status: 'Pending' | 'Completed' | 'Canceled'; // Defaults to 'Pending'
  paymentMethod: 'Card' | 'Cash' | 'Online' | 'GiftCard'; // Payment method
  deliveryMethod: 'Pickup' | 'Delivery'; // Delivery method
  deliveryAddress?: string; // Optional delivery address, max length 255
  orderNotes?: string; // Optional notes for the order, max length 255
  //user?: User | null; // Optional navigation property for the associated user
}

